clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java

```diff
@@ -279,8 +278,10 @@ private static boolean maybeBeginServerReauthentication(KafkaChannel channel, Ne
         return false;
     }
 
-    private String id() {
-        return "connection-" + idGenerator.getAndIncrement();
+    private String id(SocketChannel channel) {
+        String connectionId = ConnectionIdHelper.generateConnectionId(channel.socket(), nextConnectionIndex.getAndIncrement());
+        nextConnectionIndex.compareAndSet(Integer.MAX_VALUE, 0);
```

Comment:
this protection may be a bit verbose since it is used by testing only. Also, we can inline it